[ULP-1730] Added documentation for Send Phone Message Hook (#8805)
auth0/docs: articles/hooks/extensibility-points/index.md

Extensibility points are places in the Auth0 platform where [Hooks](/hooks) can be executed.

> Depending on the extensibility point, you can use Hooks with [Database Connections](/connections/database), [Passwordless Connections](/connections/passwordless) and/or [MFA](/multifactor-authentication).

Reviewer: ```suggestion
Depending on the extensibility point, you can use Hooks with [Database Connections](/connections/database), [Passwordless Connections](/connections/passwordless) and/or [MFA](/mfa).
```

| [Client Credentials Exchange](/hooks/extensibility-points/client-credentials-exchange) | `credentials-exchange` | Database, Passwordless | Change the <dfn data-key="scope">scopes</dfn> and add custom claims to access tokens issued by the Auth0 API's `POST /oauth/token` endpoint. |
| [Pre User Registration](/hooks/extensibility-points/pre-user-registration) | `pre-user-registration` | Database, Passwordless | Prevent user registration and add custom metadata to a newly-created user. |
| [Post User Registration](/hooks/extensibility-points/post-user-registration) | `post-user-registration` | Database, Passwordless | Implement custom actions that execute asynchronously from the Auth0 authentication process after a new user registers and is added to the database. |
| [Post Change Password](/hooks/extensibility-points/post-change-password) | `post-change-password` | Database | Implement custom actions to be executed after a successful user password change. |

> | [Send Phone Message](/hooks/extensibility-points/send-phone-message) | `send-phone-message` | MFA | Implement a custom SMS gateway that executes when a User is prompted for an MFA code. |

Reviewer 2: ```suggestion
| [Send Phone Message](/hooks/extensibility-points/send-phone-message) | `send-phone-message` | MFA | Implement a custom SMS gateway to deliver MFA one-time-use codes |
```

| [Client Credentials Exchange](/hooks/extensibility-points/client-credentials-exchange) | `credentials-exchange` | Database, Passwordless | Change the <dfn data-key="scope">scopes</dfn> and add custom claims to access tokens issued by the Auth0 API's `POST /oauth/token` endpoint. |
| [Pre User Registration](/hooks/extensibility-points/pre-user-registration) | `pre-user-registration` | Database, Passwordless | Prevent user registration and add custom metadata to a newly-created user. |
| [Post User Registration](/hooks/extensibility-points/post-user-registration) | `post-user-registration` | Database, Passwordless | Implement custom actions that execute asynchronously from the Auth0 authentication process after a new user registers and is added to the database. |
| [Post Change Password](/hooks/extensibility-points/post-change-password) | `post-change-password` | Database | Implement custom actions to be executed after a successful user password change. |

> | [Send Phone Message](/hooks/extensibility-points/send-phone-message) | `send-phone-message` | MFA | Implement a custom SMS or Voice call gateway that executes when a User is prompted for an MFA code. |

Reviewer 2: ```suggestion
| [Send Phone Message](/hooks/extensibility-points/send-phone-message) | `send-phone-message` | MFA | Implement a custom SMS gateway that executes when a User is prompted for an MFA code. |
```
